python/tvm/ir/instrument.py

```diff
@@ -255,3 +255,18 @@ def render():
                 profiles = timing_inst.render()
         """
         return _ffi_instrument_api.RenderTimePassProfiles()
+
+
+@pass_instrument
+class PassPrintingInstrument:
+    def __init__(self, print_before_pass_name, print_after_pass_name):
+        self.print_before_pass_name = print_before_pass_name
+        self.print_after_pass_name = print_after_pass_name
+
+    def run_before_pass(self, mod, pass_info):
+        if pass_info.name in self.print_before_pass_name:
+            print(f"Print IR before:\n{pass_info.name}\n{mod}\n\n")
```

Comment:
nit: I feel it's slightly easier to read if "Print IR before: pass-name" is on the same line. Happy either way though